conference/talk_voting.py

```diff
@@ -0,0 +1,125 @@
+from django.conf.urls import url
+from django.contrib.auth.decorators import login_required
+from django.db.models import Q, Prefetch
+from django.shortcuts import get_object_or_404
+from django.template.response import TemplateResponse
+
+from conference.models import Conference, Talk, VotoTalk
+
+
+@login_required
+def talk_voting(request):
+
+    current_conference = Conference.objects.current()
+
+    if not current_conference.voting():
+        return TemplateResponse(
+            request, "ep19/bs/talk_voting/voting_is_closed.html"
+        )
+
+    if not is_user_allowed_to_vote(request.user):
+        return TemplateResponse(
+            request, "ep19/bs/talk_voting/voting_is_unavailable.html", {
+                'conference': current_conference,
+            }
+        )
+
+    talks = (
+        Talk.objects.filter(
+            Q(conference=current_conference.code) & ~Q(created_by=request.user)
+        )
+        .order_by("?")
+        .prefetch_related(
+            Prefetch(
+                "vototalk_set",
+                queryset=VotoTalk.objects.filter(user=request.user),
+                to_attr="votes",
+            )
+        )
+    )
+
+    return TemplateResponse(
+        request,
+        "ep19/bs/talk_voting/voting.html",
+        {"talks": talks, "VotingOptions": VotingOptions},
+    )
+
+
+def is_user_allowed_to_vote(user):
+    """
+    Checks if user is allowed to vote at the moment of accessing this function
+    This usually means checking if they have at least one ticket associated
+    with their account (either for this or any of the past years
+    """
+    return user.ticket_set.all().exists()
```

Comment:
I think last year it was only users who have attended the conference before could vote - just want to confirm this change in behaviour is intended.

Reply:
It's essentially "attended before or bought a ticket for this year already"

Reply:
I think last year it was the same, but if it wasn't it should have been. :)